spec/linter-luacheck-spec.js

```diff
@@ -0,0 +1,67 @@
+'use babel';
+
+import { join } from 'path';
+// eslint-disable-next-line import/no-extraneous-dependencies, no-unused-vars
+import { beforeEach, it, fit } from 'jasmine-fix';
+import linterLuacheck from '../lib/init';
+
+const emptyPath = join(__dirname, 'fixtures', 'empty.lua');
+const goodPath = join(__dirname, 'fixtures', 'good.lua');
+const warningPath = join(__dirname, 'fixtures', 'warning.lua');
+const errorPath = join(__dirname, 'fixtures', 'error.lua');
+
+const errorStr = "expected identifier near 'local'";
+const warningStr = "unused argument 'self'";
+
+describe('The luacheck provider for Linter', () => {
+  const lint = linterLuacheck.provideLinter().lint;
+
+  beforeEach(async () => {
+    atom.workspace.destroyActivePaneItem();
+    await atom.packages.activatePackage('linter-luacheck');
+  });
+
+  it('should be in the packages list', () =>
+    expect(atom.packages.isPackageLoaded('linter-luacheck')).toBe(true)
+  );
+
+  it('should be an active package', () =>
+    expect(atom.packages.isPackageActive('linter-luacheck')).toBe(true)
+  );
+
+  it('finds nothing wrong with an empty file', async () => {
+    const editor = await atom.workspace.open(emptyPath);
+    const result = await lint(editor);
+
+    expect(result.length).toEqual(0);
+  });
+
+  it('finds nothing wrong in good.lua', async () => {
+    const editor = await atom.workspace.open(goodPath);
+    const result = await lint(editor);
+
+    expect(result.length).toEqual(0);
```

Comment:
No, for a valid file we should be returning `[]` (no messages), this tells `linter` that we found nothing wrong with the file. Returning `null` would tell `linter` not to update any current results it has, meaning if a file had a problem, then was fixed and we returned `null` then `linter` would still be reporting the error since we never told it not to.